### ISSUE TITLE
machinectl remove removing nspawn files from etc

### ISSUE THREAD

**Issue by justinvreeland** [systemd/systemd#15900]
**systemd version the issue has been seen with**
> systemd 245

**Used distribution**
 >  arch

**Expected behaviour you didn't see**
 > N/A

**Unexpected behaviour you saw**
 > my nspawn file was removed from /etc/systemd/nspawn

**Steps to reproduce the problem**
Import a machine and associated nspawn file in `/etc/systemd/nspawn`  Remove the machine with `machinectl remove` and the nspawn file will be removed as well.

**Comment by poettering** [comment]
Yupp, it removes a machine fully. I think that's what people would expect, no?

**Comment by justinvreeland** [comment]
I don't think so. I don't expect anything to remove files from etc, particularly ones it didn't put there.  systemd.nspawn also says:

> Persistent settings files created and maintained by the administrator (and thus trusted) should be placed in /etc/systemd/nspawn/

Which I interpreted to mean machinectl shouldn't be removing them as the administrator maintaining them.

**Comment by bcbnz** [comment]
This is also annoying when you have built an updated image and want to replace the existing one. `machinectl import-raw name.raw` says `Image 'name' already exists` and so you have to remove it first, meaning you have to copy the nspawn settings back too.

**Comment by badrelmers** [comment]
I do not see any problem with this, I think its a wanted feature not a problem.

when I remove the machine I expect all its files are removed. it is the administrator who decided to remove the container at the end, so this is consistent:

> Persistent settings files created and maintained by the administrator (and thus trusted) should be placed in /etc/systemd/nspawn/


because it is the administrator who decided to remove the machine, and a machine means both: the image and its config files

I do not not want to remove a machine which leaves its related config files, this will force me to do more cleaning!

**Comment by superboot** [comment]
I agree that this is expected behavior, not a bug.
